src/index.ts

```diff
@@ -82,7 +116,7 @@ async function run(): Promise<void> {
       required: true
     });
     const branchRegex = new RegExp(branchRegexBase, branchRegexFlags);
-    const branchCheck = branch.match(branchRegex);
+    const branchCheck = branchRegex.exec(branch);
```

Comment:
Eventually figured out that named groups are not captured using `.match` on a regex with a `g` flag. It was succeeded by `matchAll` which returns an iterator. Seemed easier to just use `.exec` instead which does return named groups on a global regex.